pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -335,6 +349,74 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
     return brokerResponse;
   }
 
+  private void handleCaseSensitivity(BrokerRequest brokerRequest) {
+    String inputTableName = brokerRequest.getQuerySource().getTableName();
+    String actualTableName = _tableCache.getActualTableName(inputTableName);
+    brokerRequest.getQuerySource().setTableName(actualTableName);
+    //fix columns
+    if (brokerRequest.getFilterSubQueryMap() != null) {
+      Collection<FilterQuery> values = brokerRequest.getFilterSubQueryMap().getFilterQueryMap().values();
+      for (FilterQuery filterQuery : values) {
+        if (filterQuery.getNestedFilterQueryIdsSize() == 0) {
```

Comment:
I don't understand this if check. FilterQueryMap is flattened and contained all the FilterQuery. We just need to iterate on all these FilterQuery which are stored in "values" and that will take care of fixing the column name for the entire tree

Reply:
we just need to check the leaf nodes in the filtertree. this check ensures that we dont process intermediate nodes like (AND/OR)

Reply:
Got it.